cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -221,22 +219,22 @@ private EventQueue getEventQueue(String accountId, String streamName) throws IOE
     createLock.lock();
     try {
       // Double check
-      eventQueue = eventQueues.get(streamName);
+      eventQueue = eventQueues.get(streamId);
       if (eventQueue != null) {
         return eventQueue;
       }
 
-      if (!streamMetaStore.streamExists(accountId, streamName)) {
+      if (!streamMetaStore.streamExists(streamId.getNamespaceId(), streamName)) {
```

Comment:
Why the metastore not taking `Id.Stream` but rather need to have namespace and stream name separated?